exporter/elasticsearchexporter/bulkindexer.go

```diff
@@ -72,7 +72,11 @@ func bulkIndexerConfig(client esapi.Transport, config *Config) docappender.BulkI
 	}
 	var compressionLevel int
 	if config.Compression == configcompression.TypeGzip {
-		compressionLevel = gzip.BestSpeed
+		if config.CompressionParams.Level == 0 {
+			compressionLevel = gzip.BestSpeed
```

Comment:
Q: can we move this default to createDefaultConfig? (I know I wrote that code but in retrospect I should have moved this default compression level to createDefaultConfig, maybe it is a good time now)